File: src/main/java/com/github/aivancioglo/resttest/setters/OAuth1TokenSecret.java
package com.github.aivancioglo.resttest.setters;

import com.github.aivancioglo.resttest.http.HTTPRequest;

public class OAuth1TokenSecret<T extends HTTPRequest> implements Setter<T> {
    private String tokenSecret;

    public OAuth1TokenSecret(String tokenSecret) {
        this.tokenSecret = tokenSecret;
    }

    /**
     * Update request.
     * @param request actual request.
     */
    @Override
    public void update(T request) {
        request.setOAuth1TokenSecret(tokenSecret);
    }
}

File: src/main/kotlin/com/github/aivancioglo/resttest/setters/OAuth1TokenSecret.kt
package com.github.aivancioglo.resttest.setters

import com.github.aivancioglo.resttest.http.HTTPRequest

class OAuth1TokenSecret<in T : HTTPRequest<*>>(private val tokenSecret: String) : Setter<T> {

    /**
     * Setting OAuth 1.0 token secret to your requestSpecification.
     *
     * @param request that will be updated.
     */
    override fun update(request: T) {
        request.oAuth1.tokenSecret = tokenSecret
    }
}